cclib/parser/gaussianparser.py

```diff
@@ -1853,46 +1871,73 @@ def natural_orbital_single_spin_parsing(inputfile, updateprogress_title):
-            while not "Sum of" in nline:
-                if has_charges:
-                    charges.append(float(nline.split()[2]))
-
-                if has_spin and has_charges:
-                    spins.append(float(nline.split()[3]))
-
-                if has_spin and not has_charges:
-                    spins.append(float(nline.split()[2]))
-
-                nline = next(inputfile)
-
-            if "Mulliken" in line:
-                if has_charges:
-                    self.atomcharges["mulliken"] = charges
-                if has_spin:
-                    self.atomspins["mulliken"] = spins
-
-            elif "Lowdin" in line:
-                self.atomcharges["lowdin"] = charges
-
+        # Spins may be included in the same section, e.g.
+        #
+        # Mulliken charges and spin densities:
+        #               1          2
+        #     1  O   -0.596188   0.019133
+        #     2  C    0.320624   0.000869
+        #
+        # APT and Lowdin charges are also displayed in this way
+        def extract_charges_spins(property):
+            """Extracts atomic charges and spin densities into self.atomcharges and self.atomspins dictionaries
+    
+            Inputs:
+                property - property type to be extracted as a string(e.g. Mulliken, Lowdin, APT)
+            """
+
+            headers = [" atomic charges:",
+            " charges:",
+            " charges with hydrogens summed into heavy atoms:",
+            " atomic charges with hydrogens summed into heavy atoms:",
+            " atomic spin densities:",
+            " charges and spin densities with hydrogens summed into heavy atoms:",
+            " charges and spin densities:"]
+            for header in headers:
+                if '{}{}'.format(property,header).lower() in line.lower():
+                    has_spin = 'spin' in line.lower()
+                    has_charges = 'charges' in line.lower()
+                    if has_charges and not hasattr(self, "atomcharges"):
+                        self.atomcharges = {}
+                    if has_spin and not hasattr(self, "atomspins"):
+                        self.atomspins = {}
+                    ones = next(inputfile)
+                    charges = []
+                    spins = []
+                    # calculate how many lines need iterating over based on whether property is summed into hydrogens or not
+                    is_sum = 'summed' in line
+                    if is_sum:
+                        n = self.natom - self.nhydrogen
+                    else:
+                        n = self.natom
+                    # iterate over each line and append values to a list based on whether they are charges or spins
+                    for i in range(n):
+                        nline = next(inputfile)
+                        while is_sum and nline.split()[1] == "H":  # some files include hydrogens in lists of summed charges with value 0.0, so these should not be recorded
```

Comment:
Can you put this comment on the line above the `while` and make sure it doesn't exceed ~100 characters (here and in general)?

Reply:
I've tidied up any comments I added across the file